djangocms_text_ckeditor/static/djangocms_text_ckeditor/ckeditor_plugins/cmsplugins/plugin.js

```diff
@@ -404,36 +404,40 @@
                 if (!that.options.delete_on_cancel && !that.unsaved_child_plugins.length) {
                     return;
                 }
-                if (that.unsaved_child_plugins.length) {
-                    e.preventDefault();
-                    CMS.API.Toolbar.showLoader();
-                    var data = {
-                        token: that.options.action_token
-                    };
-
-                    if (!that.options.delete_on_cancel) {
-                        data.child_plugins = that.unsaved_child_plugins;
-                    }
+                // if (that.unsaved_child_plugins.length) {
+                e.preventDefault();
+                CMS.API.Toolbar.showLoader();
+                var data = {
+                    token: that.options.action_token
+                };
 
-                    $.ajax({
-                        method: 'POST',
-                        url: that.options.cancel_plugin_url,
-                        data: data,
-                        // use 'child_plugins' instead of default 'child_plugins[]'
-                        traditional: true
-                    }).done(function () {
-                        CMS.API.Helpers.removeEventListener(
-                            'modal-close.text-plugin.text-plugin-' + that.options.plugin_id
-                        );
-                        opts.instance.close();
-                    }).fail(function (res) {
-                        CMS.API.Messages.open({
-                            message: res.responseText + ' | ' + res.status + ' ' + res.statusText,
-                            delay: 0,
-                            error: true
-                        });
-                    });
+                if (that.options.delete_on_cancel) {
+                    data.child_plugins = that.unsaved_child_plugins;
+                    data.child_plugins.push(that.options.plugin_id);
                 }
+
+                $.ajax({
+                    method: 'POST',
+                    headers: {
+                        'X-CSRFToken': CKEDITOR.tools.getCookie('csrftoken')
+                    },
```

Comment:
A CSRFToken was missing so the request never succeeded.

Reply:
that's weird. ckeditor uses https://github.com/divio/django-cms/blob/develop/cms/static/cms/js/modules/cms.base.js#L210 to set csrftoken for every request via `djangocms_text_ckeditor/templates/cms/plugins/text_plugin_change_form.html`

it could be that the wrong jquery is now being used

Reply:
that could also be the source of the problems (wrong jquery) cause the listener that runs the `cancelModalCallback` is attached via django-cms's jquery